crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -1096,15 +1133,15 @@ pub trait LinkedChunkListener<Item, Gap>: Send + Sync {
     type Error: Error + fmt::Debug + Send + Sync + 'static;
 
     /// A new chunk of type `ChunkContent::Items` has been inserted.
-    fn new_chunk_items(
+    async fn new_chunk_items(
```

Comment:
Same remark as before: I don't think `async` should infect the entire linked chunk API, the same way the `Result` did in the previous commit, and I would prefer those concerns to be fully separated by using a command pattern.

Reply:
The code is no more async \o/.